avoid CRAN warning

We need to avoid the use of deprecated Eigen thing in https://github.com/openpharma/mmrm/blob/main/src/utils.h#L32

see https://www.r-project.org/nosvn/R.check/r-devel-linux-x86_64-fedora-gcc/mmrm-00check.html, in particular:

```
Found the following significant warnings:
  utils.h:32:49: warning: 'Eigen::placeholders::all' is deprecated [-Wdeprecated-declarations]
```

--- a/src/utils.h
+++ b/src/utils.h
@@ -29,7 +29,7 @@ T1 subset_matrix(T1 input, T2 index1, T2 index2) {
 template <typename T1, typename T2>
 T1 subset_matrix(T1 input, T2 index1) {
   #if EIGEN_VERSION_AT_LEAST(3,4,0)
-    T1 ret = input(index1, Eigen::placeholders::all);
+    T1 ret = input(index1, Eigen::all);
   #else
     T1 ret(index1.size(), input.cols());
     for (decltype(index1.size()) i = 0; i < index1.size(); i++) {

